android/libraries/rib-base/src/main/java/com/badoo/ribs/core/routing/configuration/ConfigurationCommand.kt

```diff
@@ -0,0 +1,77 @@
+package com.badoo.ribs.core.routing.configuration
+
+import android.os.Parcelable
+import com.badoo.ribs.core.Node
+import com.badoo.ribs.core.routing.action.RoutingAction
+import com.badoo.ribs.core.routing.configuration.action.single.ActivateAction
+import com.badoo.ribs.core.routing.configuration.action.single.AddAction
+import com.badoo.ribs.core.routing.configuration.action.single.DeactivateAction
+import com.badoo.ribs.core.routing.configuration.action.multi.MultiConfigurationAction
+import com.badoo.ribs.core.routing.configuration.action.multi.SaveInstanceStateAction
+import com.badoo.ribs.core.routing.configuration.action.single.NoOpAction
+import com.badoo.ribs.core.routing.configuration.action.single.RemoveAction
+import com.badoo.ribs.core.routing.configuration.action.single.SingleConfigurationAction
+import com.badoo.ribs.core.routing.configuration.action.multi.SleepAction
+import com.badoo.ribs.core.routing.configuration.action.multi.WakeUpAction
+
+/**
+ * Represents a command to change one or more [ConfigurationContext] elements.
+ *
+ * The command object holds only the data necessary to resolve the [ConfigurationContext] that the
+ * command needs to be executed on on a logical level.
+ *
+ * Associated actions that need to be executed (resulting in [RoutingAction] and
+ * [Node] manipulations) are to be found in the associated [MultiConfigurationAction] or
+ * [SingleConfigurationAction] implementations.
+ */
+internal sealed class ConfigurationCommand<C : Parcelable> {
+
+    sealed class MultiConfigurationCommand<C : Parcelable> : ConfigurationCommand<C>() {
+
+        abstract val action: MultiConfigurationAction<C>
+
+        class Sleep<C : Parcelable> : MultiConfigurationCommand<C>() {
+            override val action: MultiConfigurationAction<C> =
```

Comment:
put in the constructor?

Reply:
This should not be overridden externally + it would introduce statefulness + then you have to support parcelizing it. I don't see why it would make any sense.